core/src/main/java/org/mule/runtime/core/internal/profiling/tracing/event/tracer/TracingCondition.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.core.internal.profiling.tracing.event.tracer;
+
+import org.mule.runtime.core.internal.profiling.tracing.event.span.InternalSpan;
+
+/**
+ * A condition that has to be met when a tracing operation is done.
+ *
+ * @since 4.5.0
+ */
+public interface TracingCondition {
+
+  TracingCondition NO_CONDITION = new TracingCondition() {
+
+    @Override
+    public void assertOnCurrentSpan(InternalSpan currentSpan) throws TracingConditionNotMetException {
```

Comment:
Can we just leave this as assertOnSpan instead of assertOnCurrentSpan?
The implementation can then pass the current Span, but this interface becomes more general.